streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -389,30 +346,27 @@ public void prepareCommit() {
     @Override
     public void postCommit() {
         switch (state()) {
+            case RESTORING:
             case RUNNING:
-                commitNeeded = false;
+            case SUSPENDED:
                 commitRequested = false;
+                commitNeeded = false;
 
-                if (!eosEnabled) {
-                    stateMgr.checkpoint(checkpointableOffsets());
+                if (state() == State.RESTORING) {
+                    writeCheckpointIfNeed();
+                } else if (!eosEnabled) { // if RUNNING, checkpoint only for non-eos
+                    writeCheckpointIfNeed();
                 }
 
-                log.debug("Committed");
-
-                break;
-
-            case RESTORING:
-                commitNeeded = false;
-                commitRequested = false;
-
-                stateMgr.checkpoint(checkpointableOffsets());
+                if (state() == State.SUSPENDED) {
+                    partitionGroup.clear();
```

Comment:
The `partitionGroup.clear` and `partitionGroup.close` are interchangeable right now, should we just consolidate both?

Reply:
Good call.